src/Plugin.cpp

```diff
@@ -73,134 +83,212 @@ void Plugin::init() {
+        trajectory->second->setTargetFrameName(targetAnchor->second.mFrame);
+        trajectory->second->pSamples = settings->second.mTrail->mSamples;
+        trajectory->second->pLength  = settings->second.mTrail->mLength;
+        trajectory->second->pColor   = VistaColor(
+            settings->second.mColor.r, settings->second.mColor.g, settings->second.mColor.b);
+
+        ++trajectory;
+
+        continue;
+      }
+    }
 
-  mGuiManager->getGui()->registerCallback("trajectories.setEnablePlanetMarks",
-      "Enables or disables the rendering of points marking the position of the planets.",
-      std::function([this](bool value) { mProperties->mEnablePlanetMarks = value; }));
+    // Else delete it.
+    mSolarSystem->unregisterAnchor(trajectory->second);
+    trajectory = mTrajectories.erase(trajectory);
+  }
 
-  mGuiManager->getGui()->registerCallback("trajectories.setEnableSunFlare",
-      "Enables or disables the rendering of a glare around the sun.",
-      std::function([this](bool value) { mProperties->mEnableSunFlares = value; }));
+  // Then create all new trajectories.
+  for (auto&& settings : mPluginSettings->mTrajectories) {
+    if (settings.second.mTrail) {
+      if (mTrajectories.find(settings.first) != mTrajectories.end()) {
+        continue;
+      }
 
-  logger().info("Loading done.");
-}
+      auto parentAnchor = mAllSettings->mAnchors.find(settings.second.mTrail->mParent);
+      auto targetAnchor = mAllSettings->mAnchors.find(settings.first);
 
-////////////////////////////////////////////////////////////////////////////////////////////////////
+      if (parentAnchor == mAllSettings->mAnchors.end()) {
+        logger().warn("Cannot add trajectory for '{}': There is no parent anchor '{}' defined in "
+                      "the settings!",
+            settings.first, settings.second.mTrail->mParent);
+        continue;
+      }
 
-void Plugin::deInit() {
-  logger().info("Unloading plugin...");
+      if (targetAnchor == mAllSettings->mAnchors.end()) {
+        logger().warn(
+            "Cannot add trajectory for '{}': There is no such anchor defined in the settings!",
+            settings.first);
+        continue;
+      }
 
-  for (auto const& flare : mSunFlares) {
-    mSolarSystem->unregisterAnchor(flare);
-  }
-  for (auto const& flareNode : mSunFlareNodes) {
-    mSceneGraph->GetRoot()->DisconnectChild(flareNode.get());
-  }
+      auto [parentStartExistence, parentEndExistence] = parentAnchor->second.getExistence();
+      auto [targetStartExistence, targetEndExistence] = targetAnchor->second.getExistence();
 
-  for (auto const& trajectory : mTrajectories) {
-    mSolarSystem->unregisterAnchor(trajectory);
-  }
-  for (auto const& trajectoryNode : mTrajectoryNodes) {
-    mSceneGraph->GetRoot()->DisconnectChild(trajectoryNode.get());
-  }
+      auto trajectory = std::make_shared<Trajectory>(mPluginSettings, targetAnchor->second.mCenter,
```

Comment:
There is already a `trajectory` variable declared in the parent scope.

Reply:
Fixed!